src/simulator/entity_base.cpp

```diff
@@ -0,0 +1,65 @@
+//========================================================================
+//  This software is free: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License Version 3,
+//  as published by the Free Software Foundation.
+//
+//  This software is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  Version 3 in the file COPYING that came with this distribution.
+//  If not, see <http://www.gnu.org/licenses/>.
+//========================================================================
+/*!
+  \file    entity_base.cpp
+  \brief   C++ Interface: Abstract class for objects
+  \author  Yifeng Zhu, (C) 2020
+  \email   yifeng.zhu@utexas.edu
+*/
+//========================================================================
+
+#include "simulator/entity_base.h"
+
+EntityBase::EntityBase() {
+}
+
+EntityBase::~EntityBase() {
+}
+
+void EntityBase::step(const double& dt) {
```

Comment:
Since we decided to use the style guide, all function names need to start with capital letters and use a capital letter for each new word. This happens a lot. No need to change things you didn't add right now, but follow the style for anything you do add. We'll refactor the old code as we go.

Reply:
I see. I'll go through the google's one. I used cpplint, and it didn't give me that my function name problem.

Reply:
cpplint probably needs to be configured to properly handle the google style guide.